### PR TITLE
Second code review

### PR DESCRIPTION
**Pull request by Mapiarz**

### CODE REVIEW

File: avatar/Assets/scripts/DataFrame.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// class of data recieved by AvatarController
+/// </summary>
+public class DataFrame
+{
+    public HumanBodyBones limb { get; set; }
```
Comment (Mapiarz) [inline avatar/Assets/scripts/DataFrame.cs]: Property names start with upper case

File: avatar/Assets/scripts/AutomaticDataGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
+﻿using UnityEngine;
+
+// TODO: Rename to AutomaticDataGenerator
+public class AutomaticDataGenerator : MonoBehaviour, IDataSource
+{
+    Quaternion delta, generatedValue;
```
Comment (Mapiarz) [inline avatar/Assets/scripts/AutomaticDataGenerator.cs]: Fields 'delta' and 'generatedValue' are never instantiated and thus it generates a warning. To fix this, initialize those 2 fields in Start method. For example: `generatedValue = Quaternion.Euler( new Vector3( 0.0f, 0.0f, 0.5f ) );`

Jeszcze jedna uwaga: Quaternion to struct i jego domyslna wartosc, to jakis tam kat, tak jak np. domyslna wartosc dla int to 0. Gdyby Quaternion bylby klasa, to jego domyslna wartosc byla by null i odwolanie w linii 18 wyrzucilo by wyjatek.

File: avatar/Assets/scripts/AutomaticDataGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            ChangeDirection();
+        }
+
+        generatedValue.eulerAngles = generatedValue.eulerAngles + delta.eulerAngles;
+        data.rotation = generatedValue;
+    }
+
+    public DataFrame GetData()
+    {
+        return (data);
```
Comment (Mapiarz) [inline avatar/Assets/scripts/AutomaticDataGenerator.cs]: Why the bracket?

File: avatar/Assets/scripts/AutomaticDataGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            ChangeDirection();
+        }
+
+        if (generatedValue.x <= -rotationMaximumValue)
+        {
+            ChangeDirection();
+        }
+
+        generatedValue.eulerAngles = generatedValue.eulerAngles + delta.eulerAngles;
```
Comment (Mapiarz) [inline avatar/Assets/scripts/AutomaticDataGenerator.cs]: Field generatedValue is not needed. All you have to do is change line 75 to:
`data.rotation = Quaternion.Euler( data.rotation.eulerAngles + delta.eulerAngles );`

Comment (Mapiarz) [reply avatar/Assets/scripts/AutomaticDataGenerator.cs]: Apply the same rule to other scripts where there is `generatedData`

File: avatar/Assets/scripts/AvatarController.cs [hunk 5]
```diff
@@ -6,191 +6,83 @@ public class AvatarController : MonoBehaviour
...
-    protected Transform[] bones;
+    AutomaticDataGenerator generatedDataSource;
+    [SerializeField]
+    ManualDataGenerator manualDataSource;
 
-    Animator animatorComponent = null;
-    string[] data;
-    HumanBodyBones limb;
-    public bool manualFlag;  // TODO: Fix public
+    DataFrame data;
```
Comment (Mapiarz) [inline avatar/Assets/scripts/AvatarController.cs]: Why is this needed? Isn't data returned from IDataSource implementations? Could `AssignData` at line 66 accept DataFrame as a parameter?

File: avatar/Assets/scripts/AvatarController.cs [hunk 6]
```diff
@@ -6,191 +6,83 @@ public class AvatarController : MonoBehaviour
...
-        var test = currentDataSource.GetData();
-        data = dataSource.GetData().Split('|');
-
-        limb = (HumanBodyBones)int.Parse(data[0]);
-
-        eulerPosition.x = float.Parse(data[1]);
-        eulerPosition.y = float.Parse(data[2]);
-        eulerPosition.z = float.Parse(data[3]);
+        data = currentDataSource.GetData();
         AssignData();
```
Comment (Mapiarz) [inline avatar/Assets/scripts/AvatarController.cs]: Would `AssignData(currentDataSource.GetData());` work?

File: avatar/Assets/scripts/AvatarController.cs [hunk 7]
```diff
@@ -6,191 +6,83 @@ public class AvatarController : MonoBehaviour
...
     {
-        Debug.Log(ChangeOfPosition((int)limb));
-        // TODO: Add comment that explains why this method is used in the first place
-        // Przyklad: Apply rotation only if it is visible, to conserve CPU
-        if (ChangeOfPosition((int)limb))
-        {
-            bones[(int)limb].transform.localEulerAngles = eulerPosition;
-        }
+        bonesDictionary[data.limb].rotation = data.rotation;
+        Debug.Log(data.limb);
```
Comment (Mapiarz) [inline avatar/Assets/scripts/AvatarController.cs]: Remove such logs after being done with debugging. Getting hundreds of debug logs kills performance and can lead other developers not to notice other, actually important logs.

File: avatar/Assets/scripts/AvatarController.cs [hunk 8]
```diff
@@ -6,191 +6,83 @@ public class AvatarController : MonoBehaviour
...
-        Vector3 delta;
-
-        delta = eulerPosition - GetPresentRotation(joint);
-
-        delta.x = Mathf.Abs(delta.x);
-        delta.y = Mathf.Abs(delta.y);
-        delta.z = Mathf.Abs(delta.z);
-
-        if (delta.x > 0.1 || delta.y > 0.1 || delta.z > 0.1) //visible change
+        changeFlag = !changeFlag;
```
Comment (Mapiarz) [inline avatar/Assets/scripts/AvatarController.cs]: `changeFlag` is not needed. You can compare object references. Pseudocode: 
`if currentDataSource == manualDataSource then currentDataSource = generatedDataSource; else currentDataSource = manualDataSource`

File: avatar/Assets/scripts/ManualDataGenerator.cs [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+    public void SliderSetZ(float z)
+    {
+        eulerPosition.z = z;
+    }
+
+    /// <summary>
+    /// method that defines which limb is moved by the transform
+    /// </summary>
+    void SetLimbTo(HumanBodyBones limb_)
```
Comment (Mapiarz) [inline avatar/Assets/scripts/ManualDataGenerator.cs]: Why the underscore?

File: avatar/Assets/scripts/ManualDataGenerator.cs [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+    public void SetLimbRightLeg()
+    {
+        SetLimbTo(HumanBodyBones.RightUpperLeg);
+    }
+
+    public DataFrame GetData()
+    {
+        eulerQuaternion.eulerAngles = eulerPosition;
+        data.rotation = eulerQuaternion;
+        return (data);
```
Comment (Mapiarz) [inline avatar/Assets/scripts/ManualDataGenerator.cs]: Why the brackets?

File: avatar/Assets/scripts/ManualDataGenerator.cs [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+
+    public void SetLimbRightLeg()
+    {
+        SetLimbTo(HumanBodyBones.RightUpperLeg);
+    }
+
+    public DataFrame GetData()
+    {
+        eulerQuaternion.eulerAngles = eulerPosition;
+        data.rotation = eulerQuaternion;
```
Comment (Mapiarz) [inline avatar/Assets/scripts/ManualDataGenerator.cs]: Could be simplified just like in the other data generator.

File: avatar/Assets/scripts/AutomaticDataGenerator.cs (hunk 4 above)
Comment (mNiechcial) [reply avatar/Assets/scripts/AutomaticDataGenerator.cs]: applying change as "data.rotation.eulerAngles = data.rotation.eulerAngles + delta.eulerAngles; " returns error "cannot modify return value because it is not a variable", adding data.rotation by-passes problem
